### ISSUE TITLE
After upgrading to 13.0.0 RC1: "Error while retrieving shares" while trying to share (Sharing App silently disabled)

### ISSUE THREAD

**Issue by rubrik** [nextcloud/server#7831]
### Steps to reproduce
1. Switch from stable channel to beta channel, run upgrade successfully.
2. Go to a folder under Files which is on External Storage (but it really local), switch to Gallery view.
3. Click the sharing icon top-right.
4. It fails. Error is "Error while retrieving shares"

### Expected behaviour
I should get a link.

### Actual behaviour
I get the error above.

### Server configuration

**Operating system**: Ubuntu Server artful 17.10

**Web server:** nginx 1.12.1-0ubuntu2

**Database:** mariadb-server 10.1.25-1

**PHP version:** php7.1-fpm 7.1.11-0ubuntu0.17.10.1

**Nextcloud version:** Nextcloud 13.0.0 RC 1

**Updated from an older Nextcloud/ownCloud or fresh install:** upgrade from latest current stable (12.0.0.4 I think)

**Where did you install Nextcloud from:** nextcloud.com

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
# sudo -u www-data php /var/www/nextcloud/occ app:list
Enabled:
  - activity: 2.6.1
  - bruteforcesettings: 1.0.3
  - comments: 1.3.0
  - dav: 1.4.6
  - federatedfilesharing: 1.3.1
  - federation: 1.3.0
  - files: 1.8.0
  - files_external: 1.4.1
  - files_pdfviewer: 1.2.0
  - files_texteditor: 2.5.1
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - firstrunwizard: 2.2.1
  - gallery: 18.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - nextcloud_announcements: 1.2.0
  - notifications: 2.1.2
  - oauth2: 1.1.0
  - password_policy: 1.3.0
  - previewgenerator: 1.0.9
  - provisioning_api: 1.3.0
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - survey_client: 1.1.0
  - systemtags: 1.3.0
  - theming: 1.4.1
  - twofactor_backupcodes: 1.2.3
  - updatenotification: 1.3.0
  - workflowengine: 1.3.0
Disabled:
  - admin_audit
  - encryption
  - files_sharing
  - files_trashbin
  - user_external
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
# sudo -u www-data php /var/www/nextcloud/occ config:list system
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.example.com",
            "10.10.10.20"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "13.0.0.10",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "overwritehost": "cloud.example.com",
        "overwriteprotocol": "https",
        "overwrite.cli.url": "https:\/\/cloud.example.com",
        "overwritecondaddr": "^10\\.10\\.10\\.15$",
        "htaccess.RewriteBase": "\/",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Memcached",
        "memcached_servers": [
            [
                "localhost",
                11211
            ]
        ],
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\Movie"
        ],
        "updater.release.channel": "beta",
        "updater.secret": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": "",
        "loglevel": 2
    }
}
```
</details>

**Are you using external storage, if yes which one:** yes local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

</details>

### Client configuration
**Browser:** Firefox

**Operating system:** Linux

#### Web server error log
<details>
<summary>Web server error log</summary>

```
(can't find anything useful, will try again)
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
(nothing relevant)
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
n/a
```
</details>

**Comment by rubrik** [comment]
I get the same error for non-external storage too.

**Comment by rubrik** [comment]
The ability to share things from the Files view seems to have disappeared completely. It's enabled in settings. I tried sharing an individual file, and also a folder. The three dots show no sharing option. Viewing the options for a file shows no sharing option either any more.

This all worked in the stable version.

**Comment by rubrik** [comment]
Disabling/re-enabling the External Storage app does not help.

**Comment by rubrik** [comment]
BINGO! I found the problem: after the upgrade the sharing app had become disabled. If I click enable, it tells me it needs to update. A few seconds later and sharing is working again :smile:

So I think there are two issues:
* The gallery view shows a share option when it should not (hence the odd error)
* The upgrade should also upgrade the sharing app (and not silently disable it)

**Comment by MorrisJobke** [comment]
We fixed the disabling of shipped apps in #7879 and backported it to 13.0.0 RC2. I hope then your issue is not happening again.

Thanks for the report anyways 👍

**Comment by MorrisJobke** [comment]
> T-shirt time!

Up to @jospoortvliet 😉
